### PR TITLE
Torchserve support for Intel GPUs

### PR DESCRIPTION
**Pull request by krish-navulla**
## Description

This PR will enable torchserve to leverage Intel GPU devices. This adds a config ipex_gpu_enable, to allow gpu enablement in base_handler.py

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java [hunk 1]
```diff
@@ -932,13 +946,35 @@ private static int getAvailableGpu() {
...
+                List<String> list =
+                        IOUtils.readLines(process.getInputStream(), StandardCharsets.UTF_8);
+                if (list.isEmpty() || !list.get(0).contains("Device ID")) {
+                    throw new AssertionError("Unexpected xpu-smi response.");
+                }
+                for (int i = 1; i < list.size(); i++) {
+                    gpuIds.add(Integer.parseInt(list.get(i)));
+                }
+                }catch (IOException | InterruptedException e) {
+                System.out.println("xpu-smi not available or failed: " + e.getMessage());
```
Comment (lxning) [inline frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java:967]: Is Intel GPU also set [CUDA_VISIBLE_DEVICES](https://github.com/pytorch/serve/blob/master/frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java#L891-L892)? If not, this part code will not be executed.

File: ts/torch_handler/base_handler.py [hunk 2]
```diff
@@ -370,7 +377,10 @@ def inference(self, data, *args, **kwargs):
             Torch Tensor : The Predicted Torch Tensor is returned in this function.
         """
         with torch.inference_mode():
-            marshalled_data = data.to(self.device)
+            if IPEX_AVAILABLE and IPEX_GPU: # IPEX GPU
+                marshalled_data = data.to("xpu")
```
Comment (lxning) [inline ts/torch_handler/base_handler.py:381]: need add xpu id.

File: examples/intel_extension_for_pytorch/README.md [hunk 3]
```diff
@@ -90,6 +92,36 @@ Add the following lines in `config.properties` to use launcher with its default
 cpu_launcher_enable=true
 ```
 
+## TorchServe with Intel® Extension for PyTorch* and Intel GPUs
```
Comment (lxning) [inline examples/intel_extension_for_pytorch/README.md:95]: Can you please provide the information link about intel GPU performance gain?

File: frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java (hunk 1 above)
Comment (Kanya-Mo) [reply frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java:981]: They are not set with CUDA_VISIBLE_DEVICES. We have updated this part. Thank you for the suggestion!

File: ts/torch_handler/base_handler.py (hunk 2 above)
Comment (Kanya-Mo) [reply ts/torch_handler/base_handler.py]: Thank you for the suggestion. We have updated this.

File: frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java (hunk 1 above)
Comment (Kanya-Mo) [reply frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java]: They will not be set, and we've removed this part. Thanks!

File: examples/intel_extension_for_pytorch/README.md (hunk 3 above)
Comment (anupren) [reply examples/intel_extension_for_pytorch/README.md:96]: Added the intel GPU performance gain to README.md.

File: examples/intel_extension_for_pytorch/README.md [hunk 4]
```diff
@@ -90,6 +93,93 @@ Add the following lines in `config.properties` to use launcher with its default
...
+
+**Enable Intel GPU Support in TorchServe:** 
+To enable TorchServe to use Intel GPUs, set the following configuration in `config.properties`:
+   ```
+   ipex_enable=true
+   ipex_gpu_enable=true
+   ```
+To enable metric reading, additionally set:
+   ```
+   system_metrics_cmd=ts/metrics/intel_gpu_metric_collector.py --gpu ${Number of GPUs}
```
Comment (agunapal) [inline examples/intel_extension_for_pytorch/README.md:125]: Can you please move this file `intel_gpu_metric_collector.py` to this folder

**Review by agunapal — CHANGES_REQUESTED** [review]

File: examples/intel_extension_for_pytorch/README.md (hunk 4 above)
Comment (Kanya-Mo) [reply examples/intel_extension_for_pytorch/README.md]: Updated. The actual file change is also done in that metric enabling PR #3141.

**Comment by agunapal** [comment]
Hi @Kanya-Mo Can you please resolve the merge conflict. Rest of the PR looks good

**Review by agunapal — COMMENTED** [review]
LGTM

**Comment by agunapal** [comment]
Hi @Kanya-Mo Can you please format the frontend file. CI is failing because of this.

**Review by agunapal — APPROVED** [review]
LGTM

**Comment by agunapal** [comment]
Hi @Kanya-Mo Please check the lint workflow and resolve the lint issue. The PR won't merge without this passing

**Pull request merged**